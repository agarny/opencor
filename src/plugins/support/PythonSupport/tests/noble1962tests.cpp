/*******************************************************************************

Copyright (C) The University of Auckland

OpenCOR is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

OpenCOR is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program. If not, see <https://gnu.org/licenses>.

*******************************************************************************/

//==============================================================================
// Python support Noble 1962 tests
//==============================================================================

#include "../../../../tests/src/testsutils.h"

//==============================================================================

#include "noble1962tests.h"

//==============================================================================

#include <QtTest/QtTest>

//==============================================================================

void Noble1962Tests::tests()
{
    // Some tests to make sure that the Noble 1962 model works fine

    QStringList output;

    QVERIFY(!OpenCOR::runCli({ "-c", "PythonShell", OpenCOR::fileName("src/plugins/support/PythonSupport/tests/data/noble1962tests.py") }, output));
    QCOMPARE(output, OpenCOR::fileContents(OpenCOR::fileName("src/plugins/support/PythonSupport/tests/data/noble1962tests.out")));
}

//==============================================================================

QTEST_APPLESS_MAIN(Noble1962Tests)

//==============================================================================
// End of file
//==============================================================================
