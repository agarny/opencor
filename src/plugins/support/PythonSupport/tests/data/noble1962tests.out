---------------------------------------------------------------------
                          Noble 1962 model
---------------------------------------------------------------------
 - CVODE:
    - Constants:
       - membrane/Cm = [ 12.0, 12.0, 12.0, ..., 12.0, 12.0, 12.0 ]
       - sodium_channel/g_Na_max = [ 400.0, 400.0, 400.0, ..., 400.0, 400.0, 400.0 ]
       - sodium_channel/E_Na = [ 40.0, 40.0, 40.0, ..., 40.0, 40.0, 40.0 ]
       - leakage_current/g_L = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - leakage_current/E_L = [ -60.0, -60.0, -60.0, ..., -60.0, -60.0, -60.0 ]
    - States:
       - membrane/V = [ -87.0, -86.5, -86.0, ..., -39.4, -40.1, -40.8 ]
       - sodium_channel_m_gate/m = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel_h_gate/h = [ 0.8, 0.8, 0.8, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/n = [ 0.0, 0.0, 0.0, ..., 0.7, 0.7, 0.7 ]
    - Rates:
       - membrane/V/prime = [ 0.4, 0.5, 0.5, ..., -0.7, -0.7, -0.7 ]
       - sodium_channel_m_gate/m/prime = [ 0.2, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel_h_gate/h/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/n/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
    - Algebraic:
       - sodium_channel/g_Na = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel_m_gate/alpha_m = [ 0.3, 0.3, 0.3, ..., 2.0, 1.9, 1.9 ]
       - sodium_channel_h_gate/alpha_h = [ 0.1, 0.1, 0.1, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/alpha_n = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/i_Na = [ -17.8, -19.2, -19.2, ..., -38.8, -38.4, -38.0 ]
       - sodium_channel_m_gate/beta_m = [ 9.5, 9.4, 9.4, ..., 3.8, 3.9, 3.9 ]
       - sodium_channel_h_gate/beta_h = [ 0.0, 0.0, 0.0, ..., 0.6, 0.5, 0.5 ]
       - potassium_channel_n_gate/beta_n = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/g_K1 = [ 1.1, 1.1, 1.1, ..., 0.5, 0.5, 0.5 ]
       - potassium_channel/g_K2 = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - potassium_channel/i_K = [ 14.9, 15.3, 15.7, ..., 45.7, 45.5, 45.3 ]
       - leakage_current/i_Leak = [ -2.0, -2.0, -2.0, ..., 1.5, 1.5, 1.4 ]
 - Euler (forward):
    - Constants:
       - membrane/Cm = [ 12.0, 12.0, 12.0, ..., 12.0, 12.0, 12.0 ]
       - sodium_channel/g_Na_max = [ 400.0, 400.0, 400.0, ..., 400.0, 400.0, 400.0 ]
       - sodium_channel/E_Na = [ 40.0, 40.0, 40.0, ..., 40.0, 40.0, 40.0 ]
       - leakage_current/g_L = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - leakage_current/E_L = [ -60.0, -60.0, -60.0, ..., -60.0, -60.0, -60.0 ]
    - States:
       - membrane/V = [ -87.0, -86.5, -86.0, ..., -39.3, -40.0, -40.7 ]
       - sodium_channel_m_gate/m = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel_h_gate/h = [ 0.8, 0.8, 0.8, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/n = [ 0.0, 0.0, 0.0, ..., 0.7, 0.7, 0.7 ]
    - Rates:
       - membrane/V/prime = [ 0.4, 0.5, 0.5, ..., -0.7, -0.7, -0.7 ]
       - sodium_channel_m_gate/m/prime = [ 0.2, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel_h_gate/h/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/n/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
    - Algebraic:
       - sodium_channel/g_Na = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel_m_gate/alpha_m = [ 0.3, 0.3, 0.3, ..., 2.0, 1.9, 1.9 ]
       - sodium_channel_h_gate/alpha_h = [ 0.1, 0.1, 0.1, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/alpha_n = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/i_Na = [ -17.8, -19.2, -19.2, ..., -38.8, -38.4, -38.0 ]
       - sodium_channel_m_gate/beta_m = [ 9.5, 9.4, 9.4, ..., 3.8, 3.8, 3.9 ]
       - sodium_channel_h_gate/beta_h = [ 0.0, 0.0, 0.0, ..., 0.6, 0.5, 0.5 ]
       - potassium_channel_n_gate/beta_n = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/g_K1 = [ 1.1, 1.1, 1.1, ..., 0.5, 0.5, 0.5 ]
       - potassium_channel/g_K2 = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - potassium_channel/i_K = [ 14.9, 15.3, 15.7, ..., 45.8, 45.6, 45.3 ]
       - leakage_current/i_Leak = [ -2.0, -2.0, -2.0, ..., 1.6, 1.5, 1.4 ]
 - Heun:
    - Constants:
       - membrane/Cm = [ 12.0, 12.0, 12.0, ..., 12.0, 12.0, 12.0 ]
       - sodium_channel/g_Na_max = [ 400.0, 400.0, 400.0, ..., 400.0, 400.0, 400.0 ]
       - sodium_channel/E_Na = [ 40.0, 40.0, 40.0, ..., 40.0, 40.0, 40.0 ]
       - leakage_current/g_L = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - leakage_current/E_L = [ -60.0, -60.0, -60.0, ..., -60.0, -60.0, -60.0 ]
    - States:
       - membrane/V = [ -87.0, -86.5, -86.0, ..., -39.4, -40.1, -40.8 ]
       - sodium_channel_m_gate/m = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel_h_gate/h = [ 0.8, 0.8, 0.8, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/n = [ 0.0, 0.0, 0.0, ..., 0.7, 0.7, 0.7 ]
    - Rates:
       - membrane/V/prime = [ 0.4, 0.5, 0.5, ..., -0.7, -0.7, -0.7 ]
       - sodium_channel_m_gate/m/prime = [ 0.2, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel_h_gate/h/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/n/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
    - Algebraic:
       - sodium_channel/g_Na = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel_m_gate/alpha_m = [ 0.3, 0.3, 0.3, ..., 2.0, 1.9, 1.9 ]
       - sodium_channel_h_gate/alpha_h = [ 0.1, 0.1, 0.1, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/alpha_n = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/i_Na = [ -17.8, -19.2, -19.2, ..., -38.8, -38.4, -38.0 ]
       - sodium_channel_m_gate/beta_m = [ 9.5, 9.4, 9.4, ..., 3.8, 3.9, 3.9 ]
       - sodium_channel_h_gate/beta_h = [ 0.0, 0.0, 0.0, ..., 0.6, 0.5, 0.5 ]
       - potassium_channel_n_gate/beta_n = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/g_K1 = [ 1.1, 1.1, 1.1, ..., 0.5, 0.5, 0.5 ]
       - potassium_channel/g_K2 = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - potassium_channel/i_K = [ 14.9, 15.3, 15.7, ..., 45.7, 45.5, 45.3 ]
       - leakage_current/i_Leak = [ -2.0, -2.0, -2.0, ..., 1.5, 1.5, 1.4 ]
 - Runge-Kutta (2nd order):
    - Constants:
       - membrane/Cm = [ 12.0, 12.0, 12.0, ..., 12.0, 12.0, 12.0 ]
       - sodium_channel/g_Na_max = [ 400.0, 400.0, 400.0, ..., 400.0, 400.0, 400.0 ]
       - sodium_channel/E_Na = [ 40.0, 40.0, 40.0, ..., 40.0, 40.0, 40.0 ]
       - leakage_current/g_L = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - leakage_current/E_L = [ -60.0, -60.0, -60.0, ..., -60.0, -60.0, -60.0 ]
    - States:
       - membrane/V = [ -87.0, -86.5, -86.0, ..., -39.4, -40.1, -40.8 ]
       - sodium_channel_m_gate/m = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel_h_gate/h = [ 0.8, 0.8, 0.8, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/n = [ 0.0, 0.0, 0.0, ..., 0.7, 0.7, 0.7 ]
    - Rates:
       - membrane/V/prime = [ 0.4, 0.5, 0.5, ..., -0.7, -0.7, -0.7 ]
       - sodium_channel_m_gate/m/prime = [ 0.2, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel_h_gate/h/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/n/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
    - Algebraic:
       - sodium_channel/g_Na = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel_m_gate/alpha_m = [ 0.3, 0.3, 0.3, ..., 2.0, 1.9, 1.9 ]
       - sodium_channel_h_gate/alpha_h = [ 0.1, 0.1, 0.1, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/alpha_n = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/i_Na = [ -17.8, -19.2, -19.2, ..., -38.8, -38.4, -38.0 ]
       - sodium_channel_m_gate/beta_m = [ 9.5, 9.4, 9.4, ..., 3.8, 3.9, 3.9 ]
       - sodium_channel_h_gate/beta_h = [ 0.0, 0.0, 0.0, ..., 0.6, 0.5, 0.5 ]
       - potassium_channel_n_gate/beta_n = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/g_K1 = [ 1.1, 1.1, 1.1, ..., 0.5, 0.5, 0.5 ]
       - potassium_channel/g_K2 = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - potassium_channel/i_K = [ 14.9, 15.3, 15.7, ..., 45.7, 45.5, 45.3 ]
       - leakage_current/i_Leak = [ -2.0, -2.0, -2.0, ..., 1.5, 1.5, 1.4 ]
 - Runge-Kutta (4th order):
    - Constants:
       - membrane/Cm = [ 12.0, 12.0, 12.0, ..., 12.0, 12.0, 12.0 ]
       - sodium_channel/g_Na_max = [ 400.0, 400.0, 400.0, ..., 400.0, 400.0, 400.0 ]
       - sodium_channel/E_Na = [ 40.0, 40.0, 40.0, ..., 40.0, 40.0, 40.0 ]
       - leakage_current/g_L = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - leakage_current/E_L = [ -60.0, -60.0, -60.0, ..., -60.0, -60.0, -60.0 ]
    - States:
       - membrane/V = [ -87.0, -86.5, -86.0, ..., -39.4, -40.2, -40.9 ]
       - sodium_channel_m_gate/m = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel_h_gate/h = [ 0.8, 0.8, 0.8, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/n = [ 0.0, 0.0, 0.0, ..., 0.7, 0.7, 0.7 ]
    - Rates:
       - membrane/V/prime = [ 0.4, 0.5, 0.5, ..., -0.7, -0.7, -0.7 ]
       - sodium_channel_m_gate/m/prime = [ 0.2, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel_h_gate/h/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/n/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
    - Algebraic:
       - sodium_channel/g_Na = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel_m_gate/alpha_m = [ 0.3, 0.3, 0.3, ..., 2.0, 1.9, 1.9 ]
       - sodium_channel_h_gate/alpha_h = [ 0.1, 0.1, 0.1, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel_n_gate/alpha_n = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/i_Na = [ -17.8, -19.2, -19.2, ..., -38.7, -38.3, -38.0 ]
       - sodium_channel_m_gate/beta_m = [ 9.5, 9.4, 9.4, ..., 3.8, 3.9, 4.0 ]
       - sodium_channel_h_gate/beta_h = [ 0.0, 0.0, 0.0, ..., 0.6, 0.5, 0.5 ]
       - potassium_channel_n_gate/beta_n = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/g_K1 = [ 1.1, 1.1, 1.1, ..., 0.5, 0.5, 0.5 ]
       - potassium_channel/g_K2 = [ 0.0, 0.0, 0.0, ..., 0.3, 0.3, 0.3 ]
       - potassium_channel/i_K = [ 14.9, 15.3, 15.7, ..., 45.7, 45.5, 45.3 ]
       - leakage_current/i_Leak = [ -2.0, -2.0, -2.0, ..., 1.5, 1.5, 1.4 ]
